pilot/pkg/model/gateway.go

```diff
@@ -393,6 +400,67 @@ func mergeGateways(gateways []gatewayWithInstances, proxy *Proxy, ps *PushContex
 			}
 		}
 	}
+
+	if features.EnableStrictGatewayMerging {
+		filterMergedServers := func(mergedServerMap map[ServerPort]*MergedServers) {
+			// Filter servers: keep GatewayAPI servers and remove Istio Gateway servers not in GatewayAPI namespaces
+			for port, mergedServer := range mergedServerMap {
+				allowedNamespaces := sets.New[string]()
+				for _, s := range mergedServer.Servers {
+					gatewayName := gatewayNameForServer[s]
```

Comment:
I'm not sure if this is correct, should this be the namespace of the verified service account?